integration-tests/slack/slack-utils.js

```diff
@@ -0,0 +1,148 @@
+const fs = require('fs')
+const _ = require('lodash/fp')
+const fetch = require('node-fetch')
+const { parse } = require('path')
+const { postMessage } = require('./post-message')
+const { printMessageTemplate } = require('./message-templates')
+
+
+const {
+  SLACK_BOT_TOKEN: defaultToken,
+  JOB_BUILD_NUM: circleJobBuildNum
+} = process.env
+
+/**
+ * Fetch CircleCI artifact links to tests summary JSON files (were created in onRunComplete() in jest-reporter.js)
+ * @param { string } token
+ * @param { string } buildNum
+ * @returns { Promise<Array[string]> } URL to tests-summary-[0-9].json
+ */
+const fetchCircleJobArtifacts = async ({ token = defaultToken, buildNum = circleJobBuildNum } = {}) => {
+  if (!buildNum) {
+    throw new Error(`**  ERROR: Missing CircleCI build number. Failed to fetch CircleCI job artifacts.`)
+  } else if (!token) {
+    throw new Error(`**  ERROR: Missing CircleCI API token. Failed to fetch CircleCI job artifacts.`)
+  }
+
+  // Find more arguments and details of the response: https://circleci.com/docs/api/v2/index.html#operation/getJobArtifacts
+  const apiUrl = 'https://circleci.com/api/v2/project/github/DataBiosphere/terra-ui'
+  const headers = { Accept: 'application/json', Authorization: `Bearer ${token}` }
+
+  try {
+    const response = await fetch(`${apiUrl}/${buildNum}/artifacts`, { method: 'GET', headers })
+    const json = await response.json()
+    const { items } = json
+    const filtered = _.filter(item => item['path'].toString().includes('tests-summary'), items)
+    return filtered.map(item => item.url)
+  } catch (e) {
+    console.error(`**  ERROR: Encountered unexpected error when getting CircleCI build_number: ${buildNum} artifacts:`, e)
+    throw e
+  }
+}
+
+const getFailedTestFileName = aggregatedResult => {
+  const failedTests = []
+  _.forEach(testResult => {
+    if (testResult.numFailingTests > 0) {
+      failedTests.push(parse(testResult.testFilePath).name)
+    }
+  }, aggregatedResult.testResults)
+  return failedTests
+}
+
+/**
+ *
+ * @returns {Promise<Array>}
+ */
+const getFailedTests = async () => {
+  const headers = { Accept: 'application/json' }
+  const urls = await fetchCircleJobArtifacts()
+  const tests = []
+  for (const url of urls) {
+    try {
+      // Parse all tests-summary JSON for failed tests
+      const response = await fetch(url, { method: 'GET', headers })
+      const json = await response.json()
+      const failedTest = getFailedTestFileName(json)
+      tests.push(...failedTest)
```

Comment:
Nit: Naming makes this confusing. Because of the singular variable name, it wasn't obvious `failedTest` is an array. When I first read this, I assumed `failedTest` is an object and thus spreading it when calling `tests.push` didn't make sense.

```suggestion
      const failedTests = getFailedTestFileName(json)
      tests.push(...failedTests)
```

Reply:
👍🏻